ui/src/components/ToS.js

```diff
@@ -0,0 +1,134 @@
+
+import {ListItem} from '@material/mwc-list/mwc-list-item';
+ 
+/**
+ * Component responsible for providing users a means to agree
+ * to the terms of service. 
+ */
+export class ToS extends LitElement {
+
+    static get properties() {
+        return {
+            country: {type: String},
+            tos: {type: Array},
+            disabled: {type: Boolean}
+        };
+    }
+
+    static get styles() {
+        return style;
+    }
+
+    constructor() {
+        super();
+        this.tos = [];
+        this.disabled = false;
+        this.tosService = new ToSService(); 
+    }
+
+    async firstUpdated() {
+        this.tos = await this.tosService.getToS(this.country);
+    }
+
+    updated() {
+        this.detectOverflow(); 
+    }
+
+    /**
+     * Enables the 'accept-button' if it has been detected that the user has read the terms of service
+     * in its entirety. 
+     */
+    checkScroll() {
+        const termsContainer = this.shadowRoot.getElementById('terms-container');
+        if(termsContainer.offsetHeight + termsContainer.scrollTop >= termsContainer.scrollHeight - 5) {
+            this.disabled = false;
+        }
+    }
+
+    /**
+     * Disables the 'accept-button' if it has been detected that the terms container cannot 
+     * be viewed in its entirety without scrolling. 
+     */
+    detectOverflow() {
+        const termsContainer = this.shadowRoot.getElementById('terms-container');
+        if(termsContainer.clientHeight < termsContainer.scrollHeight) {
+            this.disabled = true;
+        }
+    }
+
+    /**
+     * Emits an event that causes the Terms of Service to close and the 
+     * recording page to appear. 
+     */
+    handleAcceptTerms() {
+        const event = new CustomEvent('accept-tos', {
+            bubbles: true,
+            composed: true,
+        });
+        this.dispatchEvent(event);
+    }
+
+    /**
+     * Emits an event that causes the Terms of Service to close and the Country
+     * Selector component to re-appear. 
+     */
+    handleCancelTerms() {
+        const event = new CustomEvent('cancel-tos', {
+            bubbles: true,
+            composed: true,
+        });
+        this.dispatchEvent(event);
+    }
+
+    render() {
+        return html`
+            <textarea 
+                id="terms-container"
+                class="terms"
+                dir="auto"
+                readonly
+                @scroll="${this.checkScroll}">
+                    ${this.tos}
+            </textarea>
+            <div class="button-container">
+                <mwc-button 
+                    class="cancel"
+                    unelevated 
+                    label="Cancel"
+                    @click=${this.handleCancelTerms}>
+                </mwc-button>
+                <mwc-button 
+                    id="accept-button"
+                    class="accept"
+                    unelevated 
+                    ?disabled="${this.disabled}"
```

Comment:
Does this work without the quotes?
?disabled=${this.disabled}

Reply:
same behavior without the quotes